src/main/scala/org/broadinstitute/dsde/vault/services/analysis/UpdateServiceHandler.scala

```diff
@@ -94,13 +94,13 @@ case class UpdateServiceHandler(requestContext: RequestContext, dmService: Actor
         case true =>
           if (fileCount == bossObjects.size) {
             log.debug("'X-Force-Location' Analysis update complete")
-            requestContext.complete(new Analysis(dmId.get, analysis.input, analysis.metadata, analysis.files))
+            requestContext.complete(new Analysis(analysis.id, analysis.input, analysis.metadata, Option(providedFiles)))
```

Comment:
In this class, the dmid should be the same as the analysis id. If they are not, then it should be an error.